src/sentry/static/sentry/app/views/alerts/utils/index.tsx

```diff
@@ -209,3 +209,35 @@ export function convertDatasetEventTypesToSource(
     return Datasource.ERROR;
   }
 }
+
+/**
+ * Attempt to guess the data source of a discover query
+ *
+ * @returns A new query without the datasource. null on no datasource
```

Comment:
nit: change comment to something like `a new object containing the datasource and query without datasource...`